src/Bridge/OpenAI/TokenOutputProcessor.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Bridge\OpenAI;
+
+use PhpLlm\LlmChain\Chain\Output;
+use PhpLlm\LlmChain\Chain\OutputProcessor;
+
+final class TokenOutputProcessor implements OutputProcessor
+{
+    public function processOutput(Output $output): void
+    {
+        $rawResponse = $output->response->getRawResponse();
+        if (null === $rawResponse) {
+            return;
+        }
+
+        $metadata = $output->response->getMetadata();
+
+        $metadata->add(
+            'remaining_tokens',
+            (int) $rawResponse->getHeaders(false)['x-ratelimit-remaining-tokens'][0],
+        );
+
+        $content = $rawResponse->getContent(false);
+        $content = \json_decode($content, true);
+
+        if (!\array_key_exists('usage', $content)) {
+            return;
+        }
+
+        $metadata->add('prompt_tokens', $content['usage']['prompt_tokens'] ?? null);
+        $metadata->add('completion_tokens', $content['usage']['completion_tokens'] ?? null);
+        $metadata->add('total_tokens', $content['usage']['total_tokens'] ?? null);
```

Comment:
I think it could be handy to be able to provide an array

```php
$metadata->add([
    'prompt_tokens' => ......,
]);
```

Reply:
what about `function add(string|array $value, mixed $content)` while content should be ignored then ofc.
I don't like the fuzzy syntax, but have no other ideas, maybe let's skip the idea completely for now @chr-hertel ?

Reply:
really two sided here - i agree that having both in one methods feels a bit overloaded, but also see the argument for the API usage being redundant here.

what do you think of using Symfony's `ParameterBag` as a blueprint here?
https://github.com/symfony/http-foundation/blob/7.2/ParameterBag.php#L60-L81

Anyhow, not a blocker for this PR and could easily be a follow up - i think the most important part here is the introduction :muscle: